### PR TITLE
Implement a `_simple_select_many_batch`

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: synapse/storage/_base.py [hunk 1]
```diff
@@ -629,6 +629,73 @@ def _simple_select_list_txn(self, txn, table, keyvalues, retcols):
...
+    def _simple_select_many_batch(self, table, column, iterable, retcols,
+                                  keyvalues={}, desc="_simple_select_many_batch",
+                                  batch_size=100):
+        """Executes a SELECT query on the named table, which may return zero or
+        more rows, returning the result as a list of dicts.
+
+        Filters rows by if value of `column` is in `iterable`.
+
+        Args:
+            txn : Transaction object
```
Comment (NegativeMjark) [inline synapse/storage/_base.py]: I don't think this takes a txn object.

File: synapse/storage/_base.py [hunk 2]
```diff
@@ -629,6 +629,73 @@ def _simple_select_list_txn(self, txn, table, keyvalues, retcols):
...
+            table : string giving the table name
+            column : column name to test for inclusion against `iterable`
+            iterable : list
+            keyvalues : dict of column names and values to select the rows with
+            retcols : list of strings giving the names of the columns to return
+        """
+        results = []
+
+        chunks = [iterable[i:i+batch_size] for i in xrange(0, len(iterable), batch_size)]
+        for chunk in chunks:
```
Comment (NegativeMjark) [inline synapse/storage/_base.py]: Should this loop be inside the txn?

File: synapse/storage/_base.py [hunk 3]
```diff
@@ -629,6 +629,73 @@ def _simple_select_list_txn(self, txn, table, keyvalues, retcols):
...
+            keyvalues : dict of column names and values to select the rows with
+            retcols : list of strings giving the names of the columns to return
+        """
+        sql = "SELECT %s FROM %s" % (", ".join(retcols), table)
+
+        clauses = []
+        values = []
+        clauses.append(
+            "%s IN (%s)" % (column, ",".join("?" for _ in iterable))
+        )
```
Comment (NegativeMjark) [inline synapse/storage/_base.py]: Can the iterable be empty?

File: synapse/storage/_base.py (hunk 1 above)
Comment (erikjohnston) [inline synapse/storage/_base.py]: Oh?

File: synapse/storage/_base.py (hunk 3 above)
Comment (erikjohnston) [inline synapse/storage/_base.py]: I've added some guards.

File: synapse/storage/_base.py (hunk 1 above)
Comment (NegativeMjark) [inline synapse/storage/_base.py]: Eh I thought this line was earlier in the file.

**Comment by NegativeMjark** [comment]
LGTM

File: synapse/storage/_base.py (hunk 1 above)
Comment (erikjohnston) [inline synapse/storage/_base.py]: That's ok. So did I which is why I fixed the line earlier in the file.

**Pull request merged**